test/integration/test_client.c

```diff
@@ -149,3 +189,35 @@ TEST(client, querySql, setUp, tearDown, 0, client_params)
 
 	return MUNIT_OK;
 }
+
+TEST(client, querySqlInterrupt, setUp, tearDown, 0, client_params)
+{
+	struct fixture *f = data;
+	uint32_t stmt_id;
+	uint64_t last_insert_id;
+	uint64_t rows_affected;
+	unsigned i;
+	struct rows rows;
+	bool done = true;
+	(void)params;
+	EXEC_SQL("CREATE TABLE test (n INT)", &last_insert_id, &rows_affected);
+
+	EXEC_SQL("BEGIN", &last_insert_id, &rows_affected);
+
+	PREPARE("INSERT INTO test (n) VALUES(123)", &stmt_id);
+	for (i = 0; i < 4098; i++) {
+		EXEC(stmt_id, &last_insert_id, &rows_affected);
+	}
+
+	EXEC_SQL("COMMIT", &last_insert_id, &rows_affected);
+
+	/* More than 1 response buffer will be needed to return all the rows, so
+	 * we are able to interrupt the query. */
+	QUERY_SQL_DONE("SELECT * FROM test", &rows, &done);
+	munit_assert_false(done);
+
+	clientSendInterrupt(f->client, NULL);
+	clientCloseRows(&rows);
+
+	return MUNIT_OK;
+}
```

Comment:
this seems redundant with the other test

Reply:
It tests the query_sql path instead of the query path. Memory of sqlite statements is managed differently, and for me it made sense to test both cases.

Reply:
Ah I missed that, good point